### ISSUE TITLE
What to do with integer inputs?

### ISSUE THREAD

**Issue by ivarne** [JuliaDiffEq/ODE.jl#26]
A common dilemma in Julia, is what to do when the user provides integers to a function where they probably want do do the calculation in floating point. Many users do not care about the difference between `13.` and `13`, and does not want the "wrong" result when they forget the `.`. 

Suggestions
1. Try to solve the problem in integer space, and fail with `InexactError` 99.9 % of the time
2. Convert to `float` internally, but convert back to Int before returning
3. Error
4. Silently convert `T <: Integer` to `float`, and return floating point results.

It seems to be the general opinion that ODE solvers are performance sensitive, and we want them to be type stable, both internally and externally, to improve performance. We then have to decide what we want to do when the user supplies integers to `tspan` and `y0`.

**Comment by ChrisRackauckas** [comment]
>There is no reasonable way to use integer or rational values in a numerical ODE solver except by promoting them to floating point.

Yes there is. With fixed timestep methods this is clearly possible. With adaptive methods, you just have to check your timestepping to change by rational values. One scheme for this is do reject/accept by doubling/halving the timestep.

Using integer and rational values for time can be good for ODE solvers which form the backend to DDE solvers since those are not very robust to small intervals that are due to propagation of discontinuities (this arises since 1/3 + 1/3 + 1/3 != 1 in floating point).
